middleware/proxy/proxy.go

```diff
@@ -80,11 +151,7 @@ func (p Proxy) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (
 				return dns.RcodeServerFailure, errUnreachable
 			}
 
-			atomic.AddInt64(&host.Conns, 1)
-
-			reply, backendErr := p.Client.ServeDNS(w, r, host)
-
-			atomic.AddInt64(&host.Conns, -1)
```

Comment:
Those two operations moved into the function that is being called. No real reason other than they seem to be operations on the host so I figure they belong in the method on the host itself rather than the caller.